File: workers/go/apps/registry.go
package apps

import (
	"os"

	"github.com/spf13/pflag"
	"github.com/temporalio/omes/clioptions"
	"github.com/temporalio/omes/workers/go/apps/lambda"
	"github.com/temporalio/omes/workers/go/apps/worker"
	"github.com/temporalio/omes/workers/go/harness"
)

const defaultAppName = "worker"

var registry = map[string]harness.App{
	"lambda": lambda.App,
	"worker": worker.App,
}

// Main selects the registered app before handing off to the harness.
func Main() {
	var appName string

	flags := pflag.NewFlagSet("go-worker-app", pflag.ContinueOnError)
	flags.SetInterspersed(false)
	flags.StringVar(&appName, "app", defaultAppName, "Go worker app")
	if err := flags.Parse(os.Args[1:]); err != nil {
		clioptions.BackupLogger.Fatal(err)
	}

	app, ok := registry[appName]
	if !ok {
		clioptions.BackupLogger.Fatalf("unknown Go worker app %q", appName)
	}

	os.Args = append(os.Args[:1], flags.Args()...)
	if err := harness.Run(app); err != nil {
		clioptions.BackupLogger.Fatal(err)
	}
}

File: workers/java/io/temporal/omes/apps/Registry.java
package io.temporal.omes.apps;

import io.temporal.omes.apps.helloworld.HelloWorldApp;
import io.temporal.omes.apps.worker.WorkerApp;
import io.temporal.omes.harness.Harness;
import java.util.Arrays;
import java.util.Map;

public final class Registry {
  private static final String DEFAULT_APP_NAME = "worker";
  private static final Map<String, Harness.App> REGISTRY =
      Map.of("helloworld", HelloWorldApp.APP, "worker", WorkerApp.APP);

  private Registry() {}

  public static void main(String... args) throws Exception {
    String appName = DEFAULT_APP_NAME;
    String[] harnessArgs = args;
    if (args.length >= 2 && "--app".equals(args[0])) {
      appName = args[1];
      harnessArgs = Arrays.copyOfRange(args, 2, args.length);
    }

    Harness.App app = REGISTRY.get(appName);
    if (app == null || appName.isEmpty()) {
      throw new IllegalArgumentException(String.format("unknown Java worker app %s", appName));
    }

    Harness.run(app, harnessArgs);
  }
}
